crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -902,8 +907,39 @@ impl CumulocityConverter {
             _ => {
                 // if device is unregistered register using auto-registration
                 if self.entity_store.get(&source).is_none() {
-                    let auto_registration_messages =
-                        self.entity_store.auto_register_entity(&source)?;
+                    let auto_registration_messages = match self
+                        .entity_store
+                        .auto_register_entity(&source)
+                    {
+                        Ok(auto_registration_messages) => auto_registration_messages,
+                        Err(e) => match e {
+                            entity_store::Error::NonDefaultTopicScheme(eid) => {
+                                // if its non supported enity, then cache it, log
+                                // and also send out error message on `errors` topic only at the first time.
+                                if !self.custom_entities.contains(&eid) {
+                                    if self.custom_entities.len() >= CUSTOM_ENTITY_CACHE_SIZE {
+                                        self.custom_entities.pop_front();
+                                    }
+                                    self.custom_entities.push_back(eid.clone());
+                                    warn!(
+                                    "Auto registration failed for the custom entity due to: {}, try explicitly registering the entity manually",
+                                    entity_store::Error::NonDefaultTopicScheme(eid.clone())
+                                );
+                                    // send a message over the error topic as well
+                                    registration_messages.push(self.new_error_message(
+                                        ConversionError::FromEntityStoreError(
+                                            entity_store::Error::NonDefaultTopicScheme(eid),
+                                        ),
+                                    ));
+                                    return Ok(registration_messages);
```

Comment:
This is confusing as the message sent is not a registration message.

```suggestion
                                    // send a message over the error topic
                                    return Ok(self.new_error_message(
                                        ConversionError::FromEntityStoreError(
                                            entity_store::Error::NonDefaultTopicScheme(eid),
                                        ),
                                    ));
```

Reply:
Fixed it